doc/reference/configuration/configuration_reference.rst

```diff
@@ -2889,6 +3004,40 @@ This section describes configuration parameters related to roles.
 
     **Since:** :doc:`3.0.0 </release/3.0.0>`.
 
+    Specify a role's configuration.
+    This option accepts a role name as the key and a role's configuration as the value.
+    To specify the roles of an instance, use the :ref:`roles <configuration_reference_roles>` option.
```

Comment:
I've described peculiars of the globals->group->replicaset->instance options merging above. It is relevant for `roles_cfg` as well: the behavior is not very intuitive here. We should fairly describe it.